### PR TITLE
Handle inaccessible runpath gracefully

### LINKED ISSUE #8864: Scheduler should handle OSError: [Errno 28] No space left on device gracefully
```
Traceback (most recent call last):
Unexpected exception in ensemble:
File "...site-packages/ert/ensemble_evaluator/_ensemble.py", line 270, in _evaluate_inner
result = await self._scheduler.execute(min_required_realizations)
Unexpected exception in ensemble:
File "...site-packages/ert/scheduler/scheduler.py", line 288, in execute
await self._monitor_and_handle_tasks(scheduling_tasks)
Unexpected exception in ensemble:
File "...site-packages/ert/scheduler/scheduler.py", line 251, in _monitor_and_handle_tasks
raise task_exception
Unexpected exception in ensemble:
File "...site-packages/ert/scheduler/job.py", line 147, in run
await self._submit_and_run_once(sem)
Unexpected exception in ensemble:
File "...site-packages/ert/scheduler/job.py", line 108, in _submit_and_run_once
await self.driver.submit(
Unexpected exception in ensemble:
File "...site-packages/ert/scheduler/lsf_driver.py", line 301, in submit
with tempfile.NamedTemporaryFile(
Unexpected exception in ensemble:
File "/usr/lib64/python3.8/tempfile.py", line 540, in NamedTemporaryFile
(fd, name) = _mkstemp_inner(dir, prefix, suffix, flags, output_type)
Unexpected exception in ensemble:
File "/usr/lib64/python3.8/tempfile.py", line 250, in _mkstemp_inner
fd = _os.open(file, flags, 0o600)
Unexpected exception in ensemble:
OSError: [Errno 28] No space left on device: '/.../realization-0/iter-2/.lsf_submit_8rtt8yv8.sh'
```

On OSError the job should fail, but Ert should not crash.

### LINKED ISSUE #8866: Handle missing path to runpath gracefully in scheduler
```
File "...site-packages/ert/ensemble_evaluator/_ensemble.py", line 270, in _evaluate_inner
    result = await self._scheduler.execute(min_required_realizations)
Unexpected exception in ensemble: 
  File "...site-packages/ert/scheduler/scheduler.py", line 288, in execute
    await self._monitor_and_handle_tasks(scheduling_tasks)
Unexpected exception in ensemble: 
  File "...site-packages/ert/scheduler/scheduler.py", line 251, in _monitor_and_handle_tasks
    raise task_exception
Unexpected exception in ensemble: 
  File "...site-packages/ert/scheduler/job.py", line 147, in run
    await self._submit_and_run_once(sem)
Unexpected exception in ensemble: 
  File "...site-packages/ert/scheduler/job.py", line 108, in _submit_and_run_once
    await self.driver.submit(
Unexpected exception in ensemble: 
  File "...site-packages/ert/scheduler/lsf_driver.py", line 301, in submit
    with tempfile.NamedTemporaryFile(
Unexpected exception in ensemble: 
  File "/usr/lib64/python3.8/tempfile.py", line 540, in NamedTemporaryFile
    (fd, name) = _mkstemp_inner(dir, prefix, suffix, flags, output_type)
Unexpected exception in ensemble: 
  File "/usr/lib64/python3.8/tempfile.py", line 250, in _mkstemp_inner
    fd = _os.open(file, flags, 0o600)
Unexpected exception in ensemble: 
FileNotFoundError: [Errno 2] No such file or directory: '/<DOES_NOT_EXIST>/realization-257/iter-0/.lsf_submit_uho0blrg.sh'
```

### PR DESCRIPTION
**Pull request by eivindjahren**
resolves #8866
resolves #8864

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [x] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/scheduler/job.py [hunk 1]
```diff
@@ -63,13 +63,20 @@ def __init__(self, scheduler: Scheduler, real: Realization) -> None:
...
         self.returncode: asyncio.Future[int] = asyncio.Future()
-        self._aborted = False
         self._scheduler: Scheduler = scheduler
         self._callback_status_msg: str = ""
         self._requested_max_submit: Optional[int] = None
         self._start_time: Optional[float] = None
         self._end_time: Optional[float] = None
 
+    def unschedule(self, msg: str) -> None:
+        self.state = JobState.ABORTED
```
Comment (berland) [inline src/ert/scheduler/job.py:73]: Send some information to the logger as well in this scenario?

Comment (berland) [reply src/ert/scheduler/job.py:73]: Is it possible to test if/how this information is conveyed to the GUI user?

Comment (eivindjahren) [reply src/ert/scheduler/job.py:73]: image

Comment (eivindjahren) [reply src/ert/scheduler/job.py:73]: Unfortunately just shows as waiting: 
image
We should probably fix that first. Note that this scenario also makes runpath deletion in the GUI wonky.

Comment (eivindjahren) [reply src/ert/scheduler/job.py:73]: Latest changes marks the realization as failed correctly: 
image

Comment (eivindjahren) [reply src/ert/scheduler/job.py:73]: But there is no message.

Comment (eivindjahren) [reply src/ert/scheduler/job.py:73]: Now shows with hover 
[Screencast from 2024-10-09 08-20-19.webm](https://github.com/user-attachments/assets/4ae1d4a8-acc3-4188-8244-b6515282d700)

Comment (eivindjahren) [reply src/ert/scheduler/job.py:74]: Also added a log message

**Review by berland — APPROVED** [review]
great!

**Pull request merged**